artistools/spectra/spectra.py

```diff
@@ -467,6 +467,9 @@ def get_spectrum(
         arr_tdelta = at.get_timestep_times(modelpath, loc="delta")
 
         try:
+            specdata[dirbin] = specdata[dirbin].with_columns([
+                pl.col(col).cast(pl.Float64) for col in specdata[dirbin].columns
+            ])
```

Comment:
I think you can reduce this to just .with_columns(pl.all().cast(pl.Float64)), but why not cast it at the source when read_csv is called?